NangoHQ/nango
# feat(nango-yaml): endpoint explicit definition

## Context

Contributes to https://linear.app/nango/issue/NAN-1934/specify-script-endpoint-section-in-nangoyaml

We want to introduce an `entry` props that will define the endpoint category. That means we can no longer store this in a string (or we can but it would be bad). We still need to support the old way but this PR introduce a more explicit mode for endpoint.

> [!NOTE]
> This PR is changing the internal format but should not introduce any blocking changes for customers. If something is no longer working it's a bug

### Major Changes

- Change default format of endpoint in parser and internally
It's just the string split in an object. That will allow us to add `entry` later.
```ts
{ method: HTTP_METHOD, path: string } 
```

- Introduce a way to enforce CLI version at the API level
I ended up not enforcing a very recent CLI version because we want to keep retro-compat for a few months, but at some we will be able to increase the minimum CLI version. 

### Minor

- Rename HTTP_VERB to HTTP_METHOD
- Remove a lot of dead code


## 🧪 How to test?

- Run the dashboard, go to integrations and check that nothing crashes
- Deploy stuff to your local dashboard using the old CLI `nango deploy dev` 
- Deploy stuff to your local dashboard using the new CLI `node nango/packages/cli/dist/index.js deploy dev`
